### ISSUE TITLE
Wrong decision url for secretaries

### ISSUE THREAD

**Issue by JeltevanBoheemen** [UiL-OTS-labs/ethics#88]
In the various Decision overviews (`open`, `all`, `my open` etc), a bug arises:

When two users belonging to the `GROUP_SECRETARY` group are assigned as reviewers, one of them will be directed to the form for the wrong Decision object (and correctly raise a NoPermission).

This is caused by the de-duplication logic in the various ListViews concerning Decisions. The problem is two-fold:

- [x] The if-else construction in several places is incorrect, it should check for `if proposal.pk not in decision` https://github.com/UiL-OTS-labs/ethics/blob/7b8c6bc8a51329721014b37afc88356b48a9e6fe/reviews/views.py#L46-L47

- [x] Only one decision per proposal is displayed in the list. This is fine, but a user's own decision should have precedence here.  
https://github.com/UiL-OTS-labs/ethics/blob/7b8c6bc8a51329721014b37afc88356b48a9e6fe/reviews/views.py#L48-L50

Both of these lines repeat for a number of views in `reviews/views.py`. Only the ones that have separate checks for secretaries need to be fixed.

**Comment by JeltevanBoheemen** [comment]
First item is resolved in https://github.com/UiL-OTS-labs/ethics/commit/b24f0e08bb3208a9f00a2626ff445c2d6fbf9718

**Comment by JeltevanBoheemen** [comment]
Proposed solution:
change
https://github.com/UiL-OTS-labs/ethics/blob/7b8c6bc8a51329721014b37afc88356b48a9e6fe/reviews/views.py#L49
to
```
else:
    if decisions[proposal.pk].pk < obj.pk and \
            obj.reviewer == self.request.user:
        decisions[proposal.pk] = obj
```

However, I'm unsure if this would cause unforeseen problems. I suspect there is a reason that the latest decision is picked.

**Comment by tymees** [comment]
The latest decision is picked because there are a few scenarios that might result in multiple decisions existing for the secretary 
(secretary was also the supervisor, or multiple review rounds on the same proposal). Picking the latest should result in the correct one. (If not, something else has gone horribly wrong). 

Moving onto your proposed solution, it's probably better to add a new if-statement checking for that before the unique check. Otherwise, the oldest non-current-user decision would be shown instead of the latest decision.

However, wouldn't this hide proposals entirely if only the other secretary has been given a decision? 

I think there are 3 situations:

1. The current user has a decision alongside the other secretary
2. The current user has a decision, but the other secretary does not
3. The current user doesn't have a decision, but the other secretary does
 
The solution proposed would work for 1 and 2, but not for 3. I think....

As for a different solution, I was thinking of annotating the decisions with a 'is owned by current user' boolean, and replacing the decision link with a warning to add yourself as a reviewer. In addition, we would still need to apply your solution as well, but only if the proposal has a decision for the current user. 

I'll look at implementing something now, just dumping my thoughts on it here :)

**Comment by tymees** [comment]
Implemented a fix in e191194f56539e1fdc6317b1d4b8ac892a3b2947

2 main changes:

The view-code now checks if the proposal even has a decision for the current user. If it does, it will select the latest decision for that user. Otherwise, the latest overall will be selected.

The template-code now adds a button for secretaries that are not yet in current_reviewers. This button links to a redirect view, that adds a decision for that secretary and redirects to the form. 

This does bypass the 'assign reviewers' phase, as it adds a decision after the fact. However, I've done that manually several times and it should not cause problems. 

In addition, the first issue was also present in the views that use reviews instead of decisions, so I fixed those as well

**Comment by JeltevanBoheemen** [comment]
Number 3 is a good point, didn't think of that one. I like the implementation.

The only downside I can think of: users that are BOTH secretary and committee member (current extra secretaries are) can't differentiate can't tell if need to decide because another secretary assigned them as a committee member, or in their role as extra secretary. In practise, this isn't that important though: we should encourage only one active secretary at any given time.
